bin/setup_java_project.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+function notify_failure() {
+    echo -e "\033[0;31m"
+    echo "Contact C4 by emailing us at c4@cornellcollege.edu, or through our Discord in the LED sign channel!"
+    echo "We will try to help you get it working!"
+    echo -e "Failed to setup java project!\033[0m"
+}
+
+trap notify_failure ERR
+set -o errexit
+
+mvn --version
+
+# Set current directory to the directory containing this script.
+cd "$(dirname "$0")"
+
+py4j_location=$(python -m pip show -f py4j | grep Location: | cut -d' ' -f2)
+echo "Py4j found at" $py4j_location
+
+py4j_jar_location=$(python -m pip show -f py4j | grep jar | awk '{$1=$1};1')
+echo "Relative to that directory, found py4j jar at" $py4j_jar_location
+
+cd $py4j_location
+cd $(dirname $py4j_jar_location)
+
+jar_name=$(basename $py4j_jar_location)
+
+jar_version=$(echo $jar_name | grep -oP "(?<=py4j).+(?=\.jar)")
```

Comment:
`grep -oP` errors on macOS because `-P` is not an option

looking at solutions, I have to install grep (with `brew install grep`), and then change this to `ggrep`.

Reply:
Yeah, I am thinking at this point that I just need to turn these into Python scripts to attempt to solve our compatibility problem. I should probably have known there would be a lot of differences what these commands support depending on the platform.

What directory should I place the new Python scripts in?

Reply:
I've just decided to put everything in a new directory called `tools/`, so now we have:
1. `tools/setup_java_project.py`
2. `tools/compile_java_project.py`
3. `tools/start_java_server.py`